lib/msf/core/modules/metadata/cache.rb

```diff
@@ -72,7 +87,7 @@ def get_unchanged_module_references
 
     @module_metadata_cache.each_value do |module_metadata|
 
-      unless module_metadata.path and ::File.exist?(module_metadata.path)
+      if !module_metadata.path || !::File.exist?(module_metadata.path)
```

Comment:
Why was this converted from `unless` to `if` statement? The style guide suggests "[Favor unless over if for negative conditions (or control flow ||)](https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)". Dropping the `and` keyword was in line with the style guide.

Reply:
clarity, readability.  I think its fine in simple constructs but with compound constructs unless confuses me (wasn't my code originally).  Also you quoted what seems to be a random style guide, is this by extension connection to the metasploit style guide?

Reply:
Matthew linked the "official" style guide Metasploit Framework encourages contributors to follow.  See the first bullet in the [Code Contributions section of Framework's CONTRIBUTING.md file](https://github.com/rapid7/metasploit-framework/blob/master/CONTRIBUTING.md#code-contributions).

Reply:
That said, I find unless to be confusing enough at times with just one conditional check, much less multiple conditional checks...  :P

Reply:
thanks